contracts/DebtLocker.sol

```diff
@@ -1,116 +1,61 @@
-
-        // Update DebtLocker storage variables if Loan storage variables has been updated since last claim.
-        if (newFee > 0)             lastFeePaid         = newFee;
-        if (newExcess > 0)          lastExcessReturned  = newExcess;
-        if (newAmountRecovered > 0) lastAmountRecovered = newAmountRecovered;
-
-        // Withdraw all claimable funds via LoanFDT.
-        uint256 beforeBal = IERC20(liquidityAsset).balanceOf(address(this));                 // Current balance of DebtLocker (accounts for direct inflows).
-        ILoanLike(loan).withdrawFunds();                                                     // Transfer funds from Loan to DebtLocker.
-        uint256 claimBal  = IERC20(liquidityAsset).balanceOf(address(this)).sub(beforeBal);  // Amount claimed from Loan using LoanFDT.
+    function claim() external override isPool returns (uint256[7] memory details_) {
+        // Get loan state variables we need
+        uint256 claimableFunds = IMapleLoanLike(loan).claimableFunds();
+        require(claimableFunds > uint256(0), "DL:C:NOTHING_TO_CLAIM");
 
-        // Calculate sum of all deltas, to be used to calculate portions for metadata.
-        uint256 sum = newInterest.add(newPrincipal).add(newFee).add(newExcess).add(newAmountRecovered);
+        uint256 currentPrincipalRemaining = IMapleLoanLike(loan).principal();
 
-        // Calculate payment portions based on LoanFDT claim.
-        newInterest  = _calcAllotment(newInterest,  claimBal, sum);
-        newPrincipal = _calcAllotment(newPrincipal, claimBal, sum);
+        // Determine how much of claimableFunds are principal and fees
+        uint256 principalPortion = principalRemainingAtLastClaim - currentPrincipalRemaining;
 
-        // Calculate one-time portions based on LoanFDT claim.
-        newFee             = _calcAllotment(newFee,             claimBal, sum);
-        newExcess          = _calcAllotment(newExcess,          claimBal, sum);
-        newAmountRecovered = _calcAllotment(newAmountRecovered, claimBal, sum);
+        // Send funds to pool and treasury
+        IMapleLoanLike(loan).claimFunds(claimableFunds, pool);
 
-        IERC20(liquidityAsset).safeTransfer(pool, claimBal);  // Transfer entire amount claimed using LoanFDT.
+        // Update state variables
+        principalRemainingAtLastClaim = currentPrincipalRemaining;
 
-        // Return claim amount plus all relevant metadata, to be used by Pool for further claim logic.
-        // Note: newInterest + newPrincipal + newFee + newExcess + newAmountRecovered = claimBal - dust
-        //       The dust on the right side of the equation gathers in the pool after transfers are made.
-        return([claimBal, newInterest, newPrincipal, newFee, newExcess, newAmountRecovered, newDefaultSuffered]);
+        // Set return vales
+        details_[0] = claimableFunds;
+        details_[1] = uint256(0);
+        details_[2] = principalPortion;
+        details_[3] = uint256(0);
```

Comment:
This doesn't need to be here, since it defaults to zero, but we should add a note as to why its omitted